### PR TITLE
espressif: Enable the _eve module

### PR DESCRIPTION
**Pull request by chungsoftvn-tuannguyen**
Enable the _eve module for espressif mcu.

### CODE REVIEW

**Review by dhalbert — CHANGES_REQUESTED** [review]
To do this in general, do it in `ports/espressif/mpconfigport.mk`. Right now you've enabled it on some boards, but many are omitted. But it might not fit on some boards.

**Comment by dhalbert** [comment]
We have enabled this on certain boards by request, but it's not enabled by default in general.

**Comment by chungsoftvn-tuannguyen** [comment]
The switch must be in the boards/[board_name]/mpconfigport.mk file because:
1. Some boards are already equipped with a custom display (such as [adafruit_feather_esp32s2_tft](https://learn.adafruit.com/adafruit-esp32-s2-tft-feather/overview)) without EVE chip, so user may not need _eve module there (Because _eve module only works with an EVE chip). 
2. Some boards fail to compile when the _eve module is enabled (Build CI jobs) due to an error: Too little flash (Such as [adafruit_feather_esp32c6_4mbflash_nopsram](https://circuitpython.org/board/adafruit_feather_esp32c6_4mbflash_nopsram/)).

**Comment by dhalbert** [comment]
I understand. How did you choose which boards to add `_eve_` to?

**Comment by chungsoftvn-tuannguyen** [comment]
I have selected all the boards from vendors: Adafruit, Arduino, Espressif, and AI-Thinker because these are development boards, allowing users to try them with an EVE module.

File: ports/espressif/boards/adafruit_qualia_s3_rgb666/mpconfigboard.mk [hunk 1]
```diff
@@ -14,3 +14,5 @@ CIRCUITPY_ESP_PSRAM_MODE = opi
 CIRCUITPY_ESP_PSRAM_FREQ = 80m
 
 CIRCUITPY_DOTCLOCKFRAMEBUFFER = 1
+
+CIRCUITPY__EVE = 1
```
Comment (dhalbert) [inline ports/espressif/boards/adafruit_qualia_s3_rgb666/mpconfigboard.mk:18]: This board is already for driving a particular kind of display, so let's not add `_eve` to this.

**Review by dhalbert — DISMISSED** [review]
Thanks for the discussion. I've noted one board that is designed for driving other displays and shouldn't be on this list.

@tannewt how do you feel about this addition?

**Comment by tannewt** [comment]
> @tannewt how do you feel about this addition?

I'm happy to see eve enabled on espressif.

I agree with you that the enabling should be refactored into mpconfigport. The other thing to consider is whether it is on by default for future boards.

> 1. Some boards are already equipped with a custom display (such as [adafruit_feather_esp32s2_tft](https://learn.adafruit.com/adafruit-esp32-s2-tft-feather/overview)) without EVE chip, so user may not need _eve module there (Because _eve module only works with an EVE chip).

Disable EVE on these boards instead. There is likely other display functionality disabled too.

> 2. Some boards fail to compile when the _eve module is enabled (Build CI jobs) due to an error: Too little flash (Such as [adafruit_feather_esp32c6_4mbflash_nopsram](https://circuitpython.org/board/adafruit_feather_esp32c6_4mbflash_nopsram/)).

Do something similar to: https://github.com/adafruit/circuitpython/blob/ca2a24e8d0f4dcc1b9e5fa28e029bf2ffad19dbc/ports/espressif/mpconfigport.mk#L132-L137

That way you can set a reasoned default for future boards.

**Comment by chungsoftvn-tuannguyen** [comment]
I have enabled the _eve module by default for all ESP boards with a flash size greater than 2MB as of commit 8d57867. For boards that already have a display, I have still enabled the _eve module for future use.

File: ports/espressif/boards/adafruit_qualia_s3_rgb666/mpconfigboard.mk (hunk 1 above)
Comment (chungsoftvn-tuannguyen) [reply ports/espressif/boards/adafruit_qualia_s3_rgb666/mpconfigboard.mk]: I removed this line from espressif/boards/adafruit_qualia_s3_rgb666/mpconfigboard.mk.
The configuration is refactored to espressif/mpconfigport.mk at commit https://github.com/adafruit/circuitpython/commit/8d578670be362d3f462a5dda2bd5ed6fb290c50c

**Review by tannewt — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**